src/network/networkd-route.c

```diff
@@ -2341,12 +2341,10 @@ int config_parse_multipath_route(
                 }
         }
 
-        r = ordered_set_ensure_allocated(&n->multipath_routes, NULL);
-        if (r < 0)
+        r = ordered_set_ensure_put(&n->multipath_routes, NULL, m);
+        if (r == -ENOMEM)
                 return log_oom();
-
-        r = ordered_set_put(n->multipath_routes, m);
-        if (r < 0) {
+        else if (r < 0) {
```

Comment:
doesn't matter, but you could replace "else if" by just "if" here to keep things short, given that the first branch returns from the function